overlord/snapstate/handlers.go

```diff
@@ -824,3 +824,216 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 	Set(st, snapsup.Name(), snapst)
 	return nil
 }
+
+// aliases v2
+
+func (m *SnapManager) doSetAutoAliasesV2(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+	snapsup, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		return err
+	}
+	snapName := snapsup.Name()
+	curInfo, err := snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+
+	status := snapst.AliasesStatus
+	curAliases := snapst.Aliases
+	// TODO: implement --prefer/--unaliased logic
+	newAliases, nAuto, err := refreshAliases(st, curInfo, curAliases)
+	if err != nil {
+		return err
+	}
+	if status.NoAutoYet() && nAuto != 0 {
+		status = PendingEnabledAliases
```

Comment:
Shouldn´t that be done every time there are new auto aliases, rather than just when going from zero to something?

In either case, per notes above I think we might build the above conditional just looking at the data, instead of requiring a stored state. But perhaps I'm missing something.